app/models/announcement.rb

```diff
@@ -2,53 +2,53 @@ class Announcement
   # List of valid announcement types
   TYPES = %i(warning info success danger)
 
-  # The announcement's message
-  # @return [String, nil] the announcement's message if it exists
-  attr_reader :msg
-
-  # The type of announcement
-  # @return [Symbol] the type of announcement
-  attr_reader :type
-
-  class << self
-    # Parse a file
-    # @param path [#to_s] path to file
-    # @return [Announcement] announcement object
-    def parse(path)
-      path = Pathname.new(path.to_s).expand_path
-
-      if path.file? && path.readable?
-        case path.extname
-        when ".md"
-          Announcement.new(msg: path.read)
-        when ".yml"
-          Announcement.new(YAML.safe_load(ERB.new(path.read, nil, "-").result))
-        else
-          Announcement.new
-        end
-      else
-        Announcement.new
-      end
-    end
-  end
-
-  # @param opts [#to_h] the announcement object
+  # @param opts [#to_h, #to_s] the announcement object or path
   # @option opts [#to_sym] :type (:warning) Type of announcement (:info,
   #   :success, :warning, :danger)
   # @option opts [#to_s] :msg (nil) The announcement message
   def initialize(opts = {})
-    opts = opts.to_h.symbolize_keys.compact
+    if opts.respond_to? :to_h
+      @opts = opts.to_h
+    elsif opts.respond_to? :to_s
```

Comment:
Every Ruby object responds to `to_s`. I think what you are looking for is `opts.respond_to? :to_str`.

Reply:
Unfortunately it needs to be `#to_s` because it could be initialized with a `Pathname` object. But as you point out all objects respond to `#to_s` in Ruby, so the final `else` statement is probably useless.

Reply:
An alternative is...

```ruby
if opts.respond_to? :to_h
  @opts = opts.to_h
elsif !opts.nil?
  @opts = Pathname.new(opts.to_s).expand_path
else
  @opts = {}
end
```

in the off chance someone supplies `nil` for the initialization parameter instead of leaving it blank.

Reply:
I also need to move the `#expand_path` down to the private `#opts` method in case that raises an error.